src/msssim.jl

```diff
@@ -0,0 +1,110 @@
+"""
+Computes the MS-SSIM between two images.
+
+!!! info
+
+    MSSSIM, Multi-scale SSIM is defined only for gray images. RGB images are treated as 3d Gray
+    images. General `Color3` images are converted to RGB images first, you could
+    manually expand them using `channelview` if you don't want them converted to
+    RGB first.
+```
+
+# Reference
+
+[1] Wang, Z., Bovik, A. C., Sheikh, H. R., & Simoncelli, E. P. (2004). Image quality assessment: from error visibility to structural similarity. _IEEE transactions on image processing_, 13(4), 600-612.
+
+"""
+struct MSSSIM{N} <: FullReferenceIQI
+    kernel::AbstractArray{<:Real}
+    W::NTuple{N} # number of scales inferred from length
```

Comment:
Can we make this a `NTuple{N, <:Tuple}` so that

```julia
SSIM(kernel, W) = MSSSIM(kernel, (W, ))
```
?

We could then add a constructor for `MSSSIM(kernel, W::NTuple{N, <:Real}) where N` if you feel this more convinient.

Reply:
Correct me if I'm wrong but you are talking about allowing separate values of `alpha`, `beta` and `gamma` for each scale, right?
I understand that it would unify the APIs, as right now, `W` corresponds to `(alpha, beta, gamma)` in `SSIM` but `(alpha1=beta1=gamma1, alpha2=beta2=gamma2, ...)` in `MSSSIM`. However, the paper and all of the few existing implementations assume `alpha_i=beta_i=gamma_i`, and I'm not sure if we should differ or if that may create some caveats? @johnnychen94
From the paper:
> To simplify parameter selection, we let αj=βj=γj for all j’s

Reply:
For clarification, merging the usage of `SSIM` and `MSSSIM` into one unified API would be a great improvement, but if this becomes too difficult to handle, it's okay to export both.

Reply:
I have modified `MSSSIM` to use a tuple of tuples as `W` now and added the shorthand that expands `MSSSIM(kernel, (x, y, ...))` to `MSSSIM(kernel, ((x, x, x), (y, y, y), ...)`.
Please let me know what all interfaces should be exposed: because if we unify `SSIM` and `MSSSIM`, the above shorthand can't be used, as `UNIFIED_SSIM(kernel, (x, y, z))` could mean both `SSIM(kernel, (x, y, z))` and `MSSSIM(kernel, ((x, x, x), (y, y, y), (z, z, z)))`. This may not be a big enough problem, as parameters may not be modified very often.

A bigger problem would be just using the default parameters for everything, the equivalent of `MSSSIM()`. Whether `MSSSIM()` or `SSIM()` was desired cannot be inferred from `UNIFIED_SSIM()`
@johnnychen94 

_(`UNIFIED_SSIM` does not exist yet, but given which way to go in the above situations, it should be straightforward to add)_

Reply:
Correct. I have to admit that I didn't consider this. Sorry about wasting your time to experiment with this.

Looks like it should still export both `SSIM` and `MSSSIM` for this.

Reply:
Even I realised only when I tried to experiment with it, no worries! We'd still prefer this tuple of tuples implementation of `MSSSIM` with the shorthand, right?